### ISSUE TITLE
Drop the support of Android 2.3.3?

### ISSUE THREAD

**Issue by zserge** [zserge/anvil#19]
Currently Anvil's minSdk is API level 10 (2.3.3).

Dropping it would mean losing 5.6% of devices. What API levels do you normally support in your new apps?

On the other hand, if we focus on API levels 15 and higher - we could move Anim.java into Anvil's core (it relies on Honeycomb animations).

Probably it would be better to drop v10 and add v19 (KitKat) instead? So that v15 would cover 95% of the devices, while v19 would cover 50% of the devices, the modern ones.

**Comment by corbt** [comment]
I'm definitely not a representative use case, but for the app I've built and am maintaining using Anvil I only care about API level 19+. We're just installing it on hardware that we control, and all of our devices already use Android 4.4 or higher.

By the way, I'm still very excited about Anvil and enjoying working with it. The combination of Anvil + Kotlin has allowed me to write a complex application in probably half the lines of code it would have otherwise taken. It's also drastically more maintainable than traditional Android programming.

I really hope that type-safe reactive views become more popular on Android in the future -- the power and safety I feel like I'm getting in Anvil is liberating!

**Comment by zserge** [comment]
@corbt Glad to hear back from the real Anvil user and congratulations on the first Anvil app!

Yes, most likely API level 10 will be dropped, keeping API level 15 the minimal target. I would also like to move v15, v19 and support libraries to external libraries. That would simplify import paths and will only require one extra line in build.gradle. But that will complicate Anvil core testing, and it will complicate "hacks" like padding(), tag(index, Object), typeface(String) and other bindings that are not generated from the API level, but written manually. They should be shared somehow across all API levels.

Anyway, I share you feeling about Anvil. I like that I can now easily prototype, write and maintain android apps using just vim and a terminal. I'm still mostly using Java, Kotlin compiler feels somewhat slow on my laptop, but I periodically try Kotlin for small app sketches.

Anvil development goes on, despite there hasn't been many announcements recently. The biggest recent achievement is State, which brings cool declarative animations with very little code. Also, I'm covering code with tests, and it's really easy (easier than with classic android approach). So the only big thing left is events (like for the event bus and flux app architecture).

I'm thinking about a roadmap for v1.0 when I can freeze the API. If you have any suggestions - please let me know. Also, I experience big problems with bringing the idea of Anvil to the community (most people don't accept it preferring to deal with ugly XMLs and writing spaghetti code for views). So if you have a blog or are an active social user - it would be cool if you helped to spread the word about Anvil. Or, if you have ideas about what should be covered in the documentation - suggestions are welcome. I'm not a native speaker, so it's a bit hard for me to express my thoughts. Sometimes I'm thinking about copy-pasting a few  quotes from articles like this - http://jlongster.com/Removing-User-Interface-Complexity,-or-Why-React-is-Awesome - it hits the point of react-like libraries and sounds inspiring.

So, summing up:

- Anvil should become modular to bring support libraries bindings, API level bindings without exploding the core library size.
- Anvil should get event bus support sooner or later, but I'm still working on it. It would make Java 7 users happier.
- Anvil should get more promotion somehow, because the upcoming Google Data Bindings library is a disaster (btw it uses Kotlin!). Anvil can serve better in most cases, but users are not aware of it.

**Comment by corbt** [comment]
Yes, I've been meaning to blog about my use of Anvil for some time but haven't gotten around to it. Not that I have much of a following, but at least my english is good. :) It's unfortunate that my app isn't available to the public as an example of how the library works. I'm toying with the idea of creating an HN reader or something simple like that as an example. Maybe in the future if my work quiets down.

**Comment by zserge** [comment]
Anvil is being rewritten now (see anvil-forge branch). It would be possible to leave API level 10 and still have animations and other cool stuff in other "flavours" of Anvil. The question is how to name the artifacts and versions to keep the publishing process simple (see issue #26)

**Comment by zserge** [comment]
Dropped Android 2.3.3 in Anvil 0.3.5. Added Android 5.0 in the same Anvil release.
